emily/handler/src/database/accessors.rs

```diff
@@ -201,7 +202,10 @@ pub async fn pull_and_update_deposit_with_retry(
         if update.is_unnecessary(&deposit_entry) {
             return Ok(deposit_entry);
         }
-        if !is_trusted_key && deposit_entry.status != Status::Pending {
+        let is_valid_untrusted_status_update = update.event.status == StatusEntry::Accepted
+            && (deposit_entry.status == Status::Pending
+                || deposit_entry.status == Status::Accepted);
```

Comment:
We probably want to exit early with `Ok(_)` when `update.event.status == StatusEntry::Accepted && deposit_entry.status == Status::Accepted`. Basically, we do not want to actually add a new entry to the History attribute in this case, but we don't want to error either. And we'll want to modify the docs with detailing this behavior and why.

Reply:
[a9b2df2](https://github.com/stacks-sbtc/sbtc/pull/1653/commits/a9b2df23db510a3e39a919ad2b21396a5f704602)

Reply:
I think then we can keep this early exist for both trusted and untrusted for now. If sidecar don't use it we can change it once it will use